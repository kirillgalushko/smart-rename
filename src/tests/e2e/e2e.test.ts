import { describe, it, expect, beforeEach, afterAll } from 'vitest';
import fs from 'fs-extra';
import path from 'path';
import { smartRename } from '../../index';

const TEST_RESULT_DIR_NAME = 'test_result';

describe('e2e', () => {
  const mocksDir = path.resolve(__dirname, 'mocks');
  const testResultDir = path.resolve(__dirname, TEST_RESULT_DIR_NAME);

  beforeEach(async () => {
    await fs.remove(testResultDir);
    await fs.copy(mocksDir, testResultDir);
  });

  afterAll(async () => {
    await new Promise((resolve) => setTimeout(resolve, 100));
    await fs.remove(testResultDir);
  });

  it('should execute basic functionality', async () => {
    console.log(testResultDir);
    await smartRename(testResultDir, {
      blacklist: [
        /-svgrepo-com/g,
        /-logo/g,
        /-color/g,
        /-icon/g,
        /_\- logo/g,
        /\d{1,}x\d{1,}/g,
        /\(.*?\)/g,
        /_.*/g,
        /id[^\s]*/g,
        /^-|-$/g,
      ],
      removeSpaces: true,
    });
    const resultFiles = await fs.readdir(testResultDir);
    expect(resultFiles).toMatchSnapshot();
  });
});
