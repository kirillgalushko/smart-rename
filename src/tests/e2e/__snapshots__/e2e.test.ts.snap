// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`e2e > should execute basic functionality 1`] = `
[
  "digital-ocean.svg",
  "download",
  "fatal-error-2.mp3",
  "gameloft.svg",
  "intel.svg",
  "intel.txt",
  "light-switch.mp3",
  "login.txt",
  "mattermost (2).svg",
  "mattermost.svg",
  "mouse-click-computer.txt",
  "notification-message-light-interface-sound-2.mp3",
  "pleasant-interface-alert-2-message-hint-ding-message-hint-ding.mp3",
]
`;
